cmd/query/app/handler_otlp_translator_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+)
+
+func readOTLPTraces() (interface{}, error) {
+	dat, err := os.ReadFile("./fixture/OTLP-transform.json")
+	if err != nil {
+		return nil, err
+	} else {
+		var out interface{}
+		json.Unmarshal(dat, out)
+		return out, nil
+
+	}
+}
+
+func readJaegerTraces() (interface{}, error) {
+	dat, err := os.ReadFile("./fixture/jaeger-transform.json")
+	if err != nil {
+		return nil, err
+	} else {
+		var out interface{}
+		json.Unmarshal(dat, out)
+		return out, nil
+	}
+}
+
+func TestTransformOTLPSuccess(t *testing.T) {
+	// mockReader := &spanstoremocks.Reader{}
+	withTestServer(func(ts *testServer) {
+		// make main reader return NotFound
+		var response interface{}
+		request, err := readOTLPTraces()
+		require.NoError(t, err)
+		err = postJSON(ts.server.URL+"/api/transform", request, response)
+		require.NoError(t, err)
+		corectResponse, err := readJaegerTraces()
+		require.NoError(t, err)
+		assert.Equal(t, response, corectResponse)
+	}, querysvc.QueryServiceOptions{})
+}
+
+
+func TestTransformOTLPEmptySucess(t *testing.T) {
+	// mockReader := &spanstoremocks.Reader{}
+	withTestServer(func(ts *testServer) {
+		// make main reader return NotFound
+		var response interface{}
+      var request interface{} // Keeping request empty for checking behaviour
+      err := postJSON(ts.server.URL+"/api/transform", request, response)
```

Comment:
not sure what the objective here. If no OTLP is send in the request, shouldn't it be a BadRequest error?